src/test/java/com/ericsson/ei/handlers/test/MongoDBHandlerTest.java

```diff
@@ -53,21 +58,35 @@ public void init() throws Exception {
     public void testGetDocuments() {
         ArrayList<String> documents = mongoDBHandler.getAllDocuments(dataBaseName, collectionName);
         assertTrue(documents.size() > 0);
+        dropCollection();
     }
 
     @Test
     public void testGetDocumentsOnCondition() {
         ArrayList<String> documents = mongoDBHandler.find(dataBaseName, collectionName, condition);
         assertTrue(documents.size() > 0);
+        dropCollection();
     }
 
     @Test
     public void testUpdateDocument() {
         assertTrue(mongoDBHandler.updateDocument(dataBaseName, collectionName, input, updateInput));
+        dropCollection();
     }
 
-    @After
-    public void dropCollection() {
+    @Test
+    public void testIsMongoDBServerUp() {
```

Comment:
This test gives us false sense of quality. We should not mock internals in other libraries.

You should go for connection and somehow tell mongoclient to disconnect.

Reply:
Then it should be a functional test and not unittest.

Reply:
Yes. You can leave the unit test cases you implemented as they are and add functional tests.